lib/ansible/modules/cloud/atomic/atomic_container.py

```diff
@@ -0,0 +1,220 @@
+
+
+def do_uninstall(module, name, backend):
+    args = ['atomic', 'uninstall', "--storage=%s" % backend, name]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+    module.exit_json(msg=out, changed=True)
+
+
+def do_rollback(module, name):
+    args = ['atomic', 'containers', 'rollback', name]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+    else:
+        changed = "Rolling back" in out
+        module.exit_json(msg=out, changed=changed)
+
+
+def core(module):
+    mode = module.params['mode']
+    name = module.params['name']
+    image = module.params['image']
+    rootfs = module.params['rootfs']
+    values = module.params['values']
+    backend = module.params['backend']
+    state = module.params['state']
+
+    module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C')
+    out = {}
+    err = {}
+    rc = 0
+
+    values_list = ["--set=%s" % x for x in values] if values else []
+
+    args = ['atomic', 'containers', 'list', '--no-trunc', '-n', '--all', '-f', 'backend=%s' % backend, '-f', 'container=%s' % name]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0:
+        module.fail_json(rc=rc, msg=err)
+        return
+    present = name in out
+
+    if state == 'present' and present:
+        module.exit_json(msg=out, changed=False)
+    elif (state in ['latest', 'present']) and not present:
+        do_install(module, mode, rootfs, name, image, values_list, backend)
+    elif state == 'latest':
+        do_update(module, name, image, values_list)
+    elif state == 'absent':
+        if not present:
+            module.exit_json(msg="The container is not present", changed=False)
+        else:
+            do_uninstall(module, name, backend)
+    elif state == 'rollback':
+        do_rollback(module, name)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            mode=dict(default=None, choices=['user', 'system']),
+            name=dict(default=None, required=True),
+            image=dict(default=None, required=True),
+            rootfs=dict(default=None),
+            state=dict(default='latest', choices=['present', 'absent', 'latest', 'rollback']),
+            backend=dict(default=None, required=True, choices=['docker', 'ostree']),
+            values=dict(type='list', default=[]),
+        ),
+    )
+
+    if module.params['values'] is not None and module.params['mode'] == 'default':
+        module.fail_json(msg="values is supported only with user or system mode", err=err)
+
+    # Verify that the platform supports atomic command
+    rc, out, err = module.run_command('atomic -v', check_rc=False)
+    if rc != 0:
+        module.fail_json(msg="Error in running atomic command", err=err)
+
+    try:
+        core(module)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Comment:
If you use a catchall exception handler like this then you should also provide the traceback like this:
``` python
import traceback

from ansible.module_utils._text import to_native
[...]
module.fail_json(msg='Unanticipated error running atomic: %s' % to_native(e), exception=traceback.format_exc())
```